lib/python/treadmill_aws/cli/admin/aws/instance.py

```diff
@@ -116,26 +117,68 @@ def create(
         image_id = aws_cli.admin.image_id(
             ec2_conn, sts_conn, image, image_account)
         secgroup_id = aws_cli.admin.secgroup_id(ec2_conn, secgroup)
-        subnet_id = aws_cli.admin.subnet_id(ec2_conn, subnet)
+
+        total_subnets = []
 
         if not key:
             key = metadata.instance_keys()[0]
 
-        hostnames = hostmanager.create_host(
-            ipa_client=ipa_client,
-            ec2_conn=ec2_conn,
-            image_id=image_id,
-            count=count,
-            disk=disk_size,
-            domain=ipa_domain,
-            key=key,
-            secgroup_ids=secgroup_id,
-            instance_type=size,
-            subnet_id=subnet_id,
-            role=role,
+        def run_ec2(count, subnet_id):
+            """Run EC2 instance(s)"""
+            hostnames = hostmanager.create_host(
+                ipa_client=ipa_client,
+                ec2_conn=ec2_conn,
+                image_id=image_id,
+                count=count,
+                disk=disk_size,
+                domain=ipa_domain,
+                key=key,
+                secgroup_ids=secgroup_id,
+                instance_type=size,
+                subnet_id=subnet_id,
+                role=role,
+            )
+            for hostname in hostnames:
+                click.echo(hostname)
+
+        for _ in subnet:
+            subnet_id = aws_cli.admin.subnet_id(ec2_conn, _)
+            total_subnets.append(
+                ec2client.list_subnets(ec2_conn, subnet_id.split())
+            )
+
+        for network in total_subnets:
+            filter_subnets = list(
+                filter(
+                    lambda x, net=network:
+                    x[0]['SubnetId'] == net[0]['SubnetId'], total_subnets
+                )
+            )
+
+            assert len(filter_subnets) == 1, \
```

Comment:
don't use asserts. this does not look like logic error. Raise exception.

If there are no subnets available, raise NoResourceError or something along these lines. you may need to add exception to aws.exceptions.